node/pkg/watchers/sui/watcher.go

```diff
@@ -351,35 +300,26 @@ func (e *Watcher) Run(ctx context.Context) error {
 				return ctx.Err()
 
 			default:
-				_, msg, err := ws.Read(ctx)
-				if err != nil {
-					logger.Error(fmt.Sprintf("ReadMessage: '%s'", err.Error()))
-					p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
-					suiConnectionErrors.WithLabelValues("channel_read_error").Inc()
-					return err
-				}
-
-				var res SuiEventMsg
-				err = json.Unmarshal(msg, &res)
+				dataWithEvents, err := e.getEvents()
 				if err != nil {
-					logger.Error("Failed to unmarshal SuiEventMsg", zap.String("body", string(msg)), zap.Error(err))
-					return fmt.Errorf("Failed to unmarshal SuiEventMsg, body: %s, error: %w", string(msg), err)
-				}
-				if res.Error != nil {
-					return fmt.Errorf("Bad SuiEventMsg, body: %s, error: %w", string(msg), err)
-				}
-				logger.Debug("SUI result message", zap.String("message", string(msg)), zap.Any("event", res))
-				if res.ID != nil {
-					logger.Error("Found an unexpected res.ID")
+					logger.Error(fmt.Sprintf("sui_data_pump Error: %s", err.Error()))
 					continue
 				}
-
-				if res.Params != nil && (*res.Params).Result != nil {
-					err := e.inspectBody(logger, *(*res.Params).Result, false)
+				for _, event := range dataWithEvents.Data {
```

Comment:
Rather than using `reverseArray` in `getEvents`, couldn't you just do the following here:
`
if len(dataWithEvents.Data) > 0 {
   for idx := len(dataWithEvents.Data) - 1; (idx >= 0); idx-- {
      event := dataEvents.Data[idx]
`

Sorry the formatting is so ugly!

Reply:
Changed